src/modules/pos/validator/components/StakeForm/StakeForm.js

```diff
@@ -153,11 +167,46 @@ const StakeForm = ({ t, stakes, account, isStakingTxPending, nextStep, history,
     posToken
   );
   const showEmptyState = !changedStakes.length || isStakingTxPending;
+  const onConfirm = async (formProps, transactionJSON, selectedPriority, fees) => {
+    if (!showEmptyState && moduleCommandSchemas && !isLoading) {
+      const moduleCommand = joinModuleAndCommand(transactionJSON);
```

Comment:
Can we scope the reward logic (from line 172 to 205) in a function `const rewards = getRewards()` to easily see the logic in this `onConfirm`?